lib/Doctrine/ruleset.xml

```diff
@@ -30,4 +30,151 @@
             <property name="ignoreNewlines" value="true"/>
         </properties>
     </rule>
+
+    <!-- Forbid using absolute class name references (except global ones) -->
+    <rule ref="SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly">
+        <properties>
+            <property name="allowFullyQualifiedGlobalClasses" type="boolean" value="true"/>
+            <property name="allowFullyQualifiedGlobalFunctions" type="boolean" value="true"/>
+            <property name="allowFullyQualifiedGlobalConstants" type="boolean" value="true"/>
+            <property name="allowFullyQualifiedNameForCollidingClasses" type="boolean" value="true"/>
+        </properties>
+    </rule>
+
+    <!-- Forbid empty lines around type declarations -->
+    <rule ref="SlevomatCodingStandard.Types.EmptyLinesAroundTypeBraces">
+        <properties>
+            <property name="linesCountAfterOpeningBrace" value="0"/>
+            <property name="linesCountBeforeClosingBrace" value="0"/>
+        </properties>
+    </rule>
+
+    <!-- Require space around colon in return types -->
+    <rule ref="SlevomatCodingStandard.TypeHints.ReturnTypeHintSpacing">
+        <properties>
+            <property name="spacesCountBeforeColon" value="1"/>
+        </properties>
+    </rule>
+
+    <!-- Require types to be written as natively if possible;
+         require iterable types to specify phpDoc with their content;
+         forbid useless/duplicated information in phpDoc -->
+    <rule ref="SlevomatCodingStandard.TypeHints.TypeHintDeclaration">
+        <properties>
+            <property name="enableEachParameterAndReturnInspection" value="true"/>
+            <property name="traversableTypeHints" type="array" value="Doctrine\Common\Collections\Collection"/>
+            <property
+                name="usefulAnnotations"
+                type="array"
+                value="
+                    @dataProvider,
+                    @deprecated,
+                    @expectedException,
+                    @expectedExceptionMessage,
+                    @expectedExceptionMessageRegExp,
+                    @expectedExceptionCode,
+                    @expectedDeprecation,
+                    @group,
+                    @internal,
+                    @link,
+                    @see,
+                    @since,
+                    @throws
+                "
+            />
+        </properties>
+    </rule>
+
+    <!-- Require using Throwable instead of Exception -->
+    <rule ref="SlevomatCodingStandard.Exceptions.ReferenceThrowableOnly"/>
+
+    <!-- Require presence of declare(strict_types=1) -->
+    <rule ref="SlevomatCodingStandard.TypeHints.DeclareStrictTypes">
+        <properties>
+            <property
+                name="newlinesCountBetweenOpenTagAndDeclare"
+                value="2"
+            />
+            <property
+                name="spacesCountAroundEqualsSign"
```

Comment:
Shouldn't `declare` have spaces around `=`? I thought it will follow assigments

Reply:
Not according to PSR-12 :(

Reply:
I guess there was thrilling and lengthy discussion about it?

Reply:
Not sure of previous discussion, but it came up in the review: https://groups.google.com/d/msg/php-fig/vZOpga3xoLg/w7_BE7lFCQAJ

Reply:
Let's see if they'll address it :)